src/trackers/impressionsTracker.ts

```diff
@@ -56,7 +58,13 @@ export function impressionsTrackerFactory(
           impressionsToStore.push(impression);
         }
       });
-
+      let deduped = impressions.length - impressionsToStore.length;
+      let impressionsToListener = impressions;
+      
+      if (strategy) {
```

Comment:
If the `impressionsTrackerFactory` will always receive a strategy (either debug, optimized or none), let's do this parameter required and avoid the `if` evaluator in favor of performance.